src/main/native/GCM.c

```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+       
 		if (rc == ICC_OSSL_SUCCESS) {
 
 			if (aadLen > 0) {
+				
+			    /* AAD */
+#ifdef DEBUG_GCM_DETAIL
+	   			if ( debug ) {
+    	    		gslogMessage ("DETAIL_GCM NI calling ICC_AES_GCM_DecryptUpdate for AAD\n");
+    			}
+#endif			    
+			    
 				rc = ICC_AES_GCM_DecryptUpdate(ockCtx, gcmCtx,aad, aadLen,NULL, 0,NULL, &updateAADlen);
-			}
-
-			if (rc == ICC_OSSL_SUCCESS) {
-				if (ciphertextLen > 0) {
-					rc = ICC_AES_GCM_DecryptUpdate(ockCtx, gcmCtx,NULL, 0,ciphertext + ciphertextOffset, ciphertextLen, plaintext + plaintextOffset, &updateOutlen);
-				}
-
-				// that needs to catch a hash mismatch condition
-				if ( rc == ICC_OSSL_SUCCESS ) {
-					// obtain up to last block of plaintext and provide tag to compare
-					rc = ICC_AES_GCM_DecryptFinal(ockCtx, gcmCtx,plaintext + plaintextOffset + updateOutlen, &finalOutlen,
-						ciphertext + ciphertextOffset + ciphertextLen, tagLen);
-
-					if (rc != ICC_OSSL_SUCCESS ) {
-						// entered an error condition here
-						if (rc == -1 ) {
-							// hash mismatch error
-							ockCheckStatus(ockCtx);
-							return ICC_AES_GCM_TAG_MISMATCH;
-						} else {
-							// generic error condition
-							ockCheckStatus(ockCtx);
-							return ICC_AES_GCM_CRYPTFINAL_FAILED;
-						}
-					}
-				} else {
+			   
+			
+				if ( rc != ICC_OSSL_SUCCESS ) {
 					ockCheckStatus(ockCtx);
+#ifdef DEBUG_GCM_DETAIL
+        			if ( debug ) {			
+						gslogMessage ("NI data ICC_AES_GCM_DecryptUpdate ICC_AES_GCM_CRYPTUPDATE_FAILED\n" );
+					}
+#endif					
+					
 					return ICC_AES_GCM_CRYPTUPDATE_FAILED;
 				}
-			} else {
-				ockCheckStatus(ockCtx);
-				return ICC_AES_GCM_CRYPTUPDATE_FAILED;
-			}
+				else {
+					
+#ifdef DEBUG_GCM_DETAIL
+	   				if ( debug ) {
+    	    			gslogMessage ("DETAIL_GCM NI AAD ICC_AES_GCM_DecryptUpdate Succeeded\n");
+    				}
+#endif
+				}
+             }
+             else {
+             	
+#ifdef DEBUG_GCM_DETAIL
+	   			if ( debug ) {
+    	    		gslogMessage ("DETAIL_GCM NI No AAD ICC_AES_GCM_DecryptUpdate not called \n");
+    			}
+#endif
+             	
+             }
+        
 		} else {
 			ockCheckStatus(ockCtx);
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	gslogMessage ("DETAIL_GCM NI ICC_AES_GCM_DecryptUpdate ICC_AES_GCM_INIT_FAILED \n");
+    		}
+#endif
+			
 			return ICC_AES_GCM_INIT_FAILED;
 		}
 	} else {
 		/* GCM CTX Failed - no need to free it. */
 		ockCheckStatus(ockCtx);
+		
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	            gslogMessage ("DETAIL_GCM NI gslogMessage ICC_AES_GCM_NEW_FAILED\n" );
+    		}
+#endif
+		
 		return ICC_AES_GCM_CTX_NEW_FAILED;
 	}
 	return 0;
+	
+#else
+      	return -1;
+#endif
 }
+/*============================================================================
+*
+* Class:     com_ibm_crypto_plus_provider_ock_NativeInterface
+* Method:    internal functions
+* Signature: 
+*/
+int GCM_UpdForUpdateDecrypt_core (JNIEnv *env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
```

Comment:
Need space before `*`.

Reply:
I think we should have no space after the * only before them.